### PR TITLE
feat: support `datediff` udf

### PR DESCRIPTION
**Pull request by zhanghaohit**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

### CODE REVIEW

File: hybridse/src/udf/default_udf_library.cc [hunk 1]
```diff
@@ -2025,6 +2025,11 @@ void DefaultUdfLibrary::InitTypeUdf() {
         .doc(R"(
             @brief Cast timestamp or string expression to date
 
+            Supported string style:
```
Comment (zhanghaohit) [inline hybridse/src/udf/default_udf_library.cc:2046]: check whether must > `1900`?

File: hybridse/src/udf/udf.cc [hunk 2]
```diff
@@ -874,6 +874,59 @@ void string_to_date(StringRef *str, Date *output,
...
+    if (date1 == nullptr || date2 == nullptr || date1->date_ <= 0 || date2->date_ <= 0) {
+        *is_null = true;
+        return;
+    }
+    int32_t year, month, day;
+    if (!date1->Decode(date1->date_, &year, &month, &day)) {
+        *is_null = true;
+        return;
+    }
+    boost::gregorian::date d1(year, month, day);
```
Comment (zhanghaohit) [inline hybridse/src/udf/udf.cc]: change to `absl`

File: hybridse/src/udf/default_udf_library.cc (hunk 1 above)
Comment (zhanghaohit) [reply hybridse/src/udf/default_udf_library.cc:2046]: yes. must >= 1900-01-01

**Review by dl239 — APPROVED** [review]

**Review by aceforeverd — APPROVED** [review]

**Review by tobegit3hub — COMMENTED** [review]
LGTM

**Pull request merged**